types/n3/index.d.ts

```diff
@@ -249,6 +249,7 @@ export class Store<Q_RDF extends RDF.BaseQuad = RDF.Quad, Q_N3 extends BaseQuad
     removeMatches(subject?: Term | null, predicate?: Term | null, object?: Term | null, graph?: Term | null): EventEmitter;
     deleteGraph(graph: Q_RDF['graph'] | string): EventEmitter;
     getQuads(subject: OTerm, predicate: OTerm, object: OTerm | OTerm[], graph: OTerm): Quad[];
+    readQuads(subject: OTerm, predicate: OTerm, object: OTerm | OTerm[], graph: OTerm): Generator<Quad>;
```

Comment:
```suggestion
```

The object of `readQuads` is cannot be an array AFAIK.

I also don't think we should be exposing this since `#match` now provides the same functionality and is a standard RDF/JS method.

Reply:
I think its better to expose it, but mark it as deprecated. It is in the docs for a while now and people started using it.

Reply:
> It is in the docs for a while now

Yes, because it pre-ceded the `#match` method because this library predates the RDF/JS spec.

> and people started using it.

Those people are already coping with this method not being available in the types (by ts-ignoring, type-casing etc.); so I don't see how this change benefits those users.


Either way I'll give @RubenVerborgh final call on this.

Reply:
But then it should be removed from N3.js first. I'm fine either way!

Reply:
I don't have any strong attachment to `readQuads`. The convention in N3.js is that, anything that does not start with an underscore, is part of the public API.

I would document for now, and we can indeed consider deprecation.

Reply:
@jeswr while I agree with you, it's more of an architectural discussion of N3js and I don't think a PR about typings is the place to make those decisions :) Ok if we continue this discussion in the N3js repo in context of a next release? Happy to adjust the typings for that one

Reply:
Yep; I suggest we mark it as deprecated in the next release https://github.com/rdfjs/N3.js/issues/355